File: 3438.find-valid-pair-of-adjacent-digits-in-string/solution.kt
class Solution {
    fun findValidPair(s: String): String {
        val f = s.groupingBy { it }.eachCount()
        for (i in 0 until s.lastIndex) {
            if (s[i] != s[i + 1] && f[s[i]] == s[i] - '0' && f[s[i + 1]] == s[i + 1] - '0') {
                return s.substring(i .. i + 1)
            }
        }
        return ""
    }
}

File: 3438.find-valid-pair-of-adjacent-digits-in-string/solution.go
func findValidPair(s string) string {
    var h [10]int
    for _, r := range s {
        h[int(r - '0')]++
    }
    for i := 1; i < len(s); i++ {
        if s[i] == s[i - 1] {
            continue
        }
        if a := int(s[i - 1] - '0'); h[a] != a {
            continue
        }
        if b := int(s[i] - '0'); h[b] != b {
            continue
        }
        return s[i - 1: i + 1]
    }
    return ""
}